core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala

```diff
@@ -462,10 +464,10 @@ class ContainerPool(
                 // We grab the size first so we know there has been enough delay for anything we are shutting down
                 val size = toBeRemoved.size()
                 1 to size foreach { _ =>
-                    val ci = toBeRemoved.poll()
-                    if (ci != null) { // should never happen but defensive
+                    val removeJob = toBeRemoved.poll()
+                    if (removeJob != null) { // should never happen but defensive
                         Thread.sleep(100) // serves to not hog docker lock and add slack
-                        teardownContainer(ci.container)
+                        teardownContainer(removeJob)
```

Comment:
How about wrapping it in an `Option` instead of null-checking?

```
Option(toBeRemoved.poll()).foreach { removeJob =>
    Thread.sleep(100)
    teardownContainer(removeJob)
}
```